### PR TITLE
feat(adapter): Add RGT to Tribe migration adapter [SIM-52]

### PR DESCRIPTION
**Pull request by FlattestWhite**
# Code Review Processes
## New Feature Review
Before submitting a pull request for new review, make sure the following is done:
* Design doc is created and posted here: [Insert Link]
* Code cleanliness and completeness is addressed via [guidelines](https://app.gitbook.com/@setprotocol-1/s/set/smart-contract-engineering/sc-code-review-process)

README Checks
- [] README has proper context for the reviewer to understand what the code includes, any important design considerations, and areas to pay more attention to

Code Checks
- [] Add explanatory comments. If there is complex code that requires specific context or understanding, note that in a comment
- [] Remove unncessary comments. Any comments that do not add additional context, information, etc. should be removed
- [] Add javadocs. 
- [] Scrub through the code for inconsistencies (e.g. removing extra spaces)
- [] Ensure there are not any .onlys in spec files

Broader Considerations
- [] Ensure variable, function and event naming is clear, consistent, and reflective for the scope of the code.
- [] Consider if certain pieces of logic should be placed in a different library, module

### CODE REVIEW

File: contracts/mocks/external/TribePegExchangerMock.sol [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    limitations under the License.
+
+    SPDX-License-Identifier: Apache License, Version 2.0
+*/
+
+pragma solidity 0.6.10;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "hardhat/console.sol";
```
Comment (ncitron) [inline contracts/mocks/external/TribePegExchangerMock.sol:23]: Probably want to remove the console log import

File: contracts/mocks/external/TribePegExchangerMock.sol [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+/**
+ @title Contract to exchange RGT with TRIBE post-merger
```
Comment (ncitron) [inline contracts/mocks/external/TribePegExchangerMock.sol:26]: Can we make the comment formatting of the title match up with the formatting we normally use (with the * on each line)

File: contracts/protocol/integration/wrap/RgtMigrationWrapAdapter.sol [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    )
+        public
+    {
+        pegExchanger = _pegExchanger;
+    } 
+
+    /* ============ External Getter Functions ============ */
+
+    /**
+     * Generates the calldata to migrate KNC Legacy to KNC.
```
Comment (ncitron) [inline contracts/protocol/integration/wrap/RgtMigrationWrapAdapter.sol:53]: Update this comment

**Review by ncitron — COMMENTED** [review]
Just a couple of nitpicks but everything looks about right. @controtie can we get this reviewed by Set?

File: contracts/mocks/external/TribePegExchangerMock.sol (hunk 1 above)
Comment (FlattestWhite) [reply contracts/mocks/external/TribePegExchangerMock.sol]: whoops - done

File: contracts/mocks/external/TribePegExchangerMock.sol (hunk 2 above)
Comment (FlattestWhite) [reply contracts/mocks/external/TribePegExchangerMock.sol]: yep done

File: contracts/protocol/integration/wrap/RgtMigrationWrapAdapter.sol (hunk 3 above)
Comment (FlattestWhite) [reply contracts/protocol/integration/wrap/RgtMigrationWrapAdapter.sol]: done

File: contracts/protocol/integration/wrap/RgtMigrationWrapAdapter.sol [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+    SPDX-License-Identifier: Apache License, Version 2.0
+*/
+
+pragma solidity 0.6.10;
+pragma experimental "ABIEncoderV2";
+
+/**
+  * @title RgtMigrationWrapAdater
+  * @author Set Protocol
```
Comment (pblivin0x) [inline contracts/protocol/integration/wrap/RgtMigrationWrapAdapter.sol:24]: @FlattestWhite should get credit for this

File: contracts/mocks/external/TribePegExchangerMock.sol [hunk 5]
```diff
@@ -49,11 +48,8 @@ contract TribePegExchangerMock {
     function exchange(uint256 amount) public {
         uint256 tribeOut = amount * exchangeRate / scalar;
         IERC20(rgt).safeTransferFrom(msg.sender, address(this), amount);
-        console.log("transferred from rgt");
-        console.log("transferring %s", tribeOut);
         console.log(IERC20(tribe).balanceOf(address(this)));
```
Comment (pblivin0x) [inline contracts/mocks/external/TribePegExchangerMock.sol:51]: let's remove this `console.log` as well

**Comment by pblivin0x** [comment]
Noting some details for the PegExchanger 
- [Contract](https://github.com/fei-protocol/fei-protocol-core/blob/develop/contracts/merger/PegExchanger.sol)
- [Mainnet address](https://github.com/fei-protocol/fei-protocol-core/blob/4d5284d3139b1cf85518867f1342d1ca4108be64/protocol-configuration/mainnetAddresses.ts#L1791): `0xc09BB5ECf865e6f69Fe62A43c27f036A426909f7`

**Comment by pblivin0x** [comment]
LGTM 💯

**Review by pblivin0x — APPROVED** [review]
LGTM 💯

**Pull request merged**